### ISSUE TITLE
Is it correct to emulate DSAC++ in this specific way?

### ISSUE THREAD

**Issue by XYZ-99** [vislearn/esac#6]
To emulate DSAC++, for example on 7Scenes, first I need to modify the `env_list.txt` to
```
../../datasets/7scenes_chess -0.006378 -0.158068 1.608667
```
, and train with:
```bash
/environments/7scenes> python ../../code/init_gating.py -c 1
/environments/7scenes> python ../../code/init_expert.py -e 0 -c 1
/environments/7scenes> python ../../code/train_esac.py -c 1
```
, and test with:
```bash
/environments/7scenes> python ../../code/test_esac.py -c 1
```
and repeat the procedure iteratively with all the 7 scenes in the 7Scenes dataset in order to obtain the performance of the pytorch-implemented DSAC++ on 7Scenes.

Is it correct?

Or can I simply modify the `env_list.txt` in the way described above(to only one line in each test) and use the pretrained model to test directly:
```bash
/environments/7scenes> python ../../code/test_esac.py -sid pretrained -c 1
```

**Comment by ebrach** [comment]
Hi,
you could use the pre-trained model with the original env_list.txt (7 entries) and test with the -os option. This option disables gating and always selects the correct expert for each scene. However, this is not 100% equivalent to DSAC++ since the end-to-end training stage optimised for a different objective. In practice, the accuracy will be close to what DSAC++ would give you (probably a bit worse).

If you want to replicate DSAC++ exactly, then the procedure in the first part of your message is the way to go. However, you have to refine the expert as well, ie. run  ref_expert.py after init_expert.py.

Note: If you want to compare to DSAC++ e.g. within a publication, bear in mind that this code (ESAC) uses a different network architecture than the original DSAC++ code and publication. The difference between the two architectures is discussed in the ESAC paper, both architectures give similar accuracy.

Depending on your use case, you might also want to take a look at our DSAC* code:  https://github.com/vislearn/dsacstar
DSAC* is an improved version of DSAC++ and comes e.g. with shorter training times.

Best,
Eric
